### DISCUSSION TITLE
Query output into dictionary

### DISCUSSION THREAD

**Discussion by michalchrzastek** [sqlalchemy/sqlalchemy#9342] [Usage Questions]
Hello,

Prior to upgrade to 2.0.x, I used it like so:

`res = db.session.query(...)`
`return [dict(r) for r in res]`
as per example here [https://github.com/sqlalchemy/sqlalchemy/discussions/7926](https://github.com/sqlalchemy/sqlalchemy/discussions/7926)

After the upgrade, the above results in a TypeError
`TypeError: cannot convert dictionary update sequence element #0 to a sequence`

So I am generating the dict like so:
`my_dict = []`
`for r in res:`
`    my_dict.append({'col1':r[0],'col2':r[1]})`
`return my_dict`

1. Why the initial method does no longer work?
2. Is the new way, the best approach?

Thanks

**Comment by CaselIT** [comment]
Hi,

I'm assuming `...` in `query()` is not an ORM model, since query would otherwise return a list of instance of that node.
If it instead returns rows instances, you can call `_asdict()` on the row to get a dict object.

documentation is here https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row._asdict
